### PR TITLE
History fragment

### PR DESCRIPTION
**Pull request by bensouthern**
This is pretty much done, still WIP.  I still have to add functionality to the buttons but that is all

### CODE REVIEW

File: app/src/main/java/com/packants/app/packants/Fragments/HistoryFragment.java [hunk 1]
```diff
@@ -24,13 +45,115 @@ public HistoryFragment() {
...
+    private void prepareListData() {
+        int histListViewWidth = histListView.getWidth();
+        listDataHeader = new ArrayList<String>();
+        listDataChild = new HashMap<String, List<String>>();
+
+        // Adding child data
+        changeFilters();
+         // Header, Child data
+    }
+    private void populateFiters() {
```
Comment (0xABC) [inline app/src/main/java/com/packants/app/packants/Fragments/HistoryFragment.java]: typo “fiters"

File: app/build.gradle [hunk 2]
```diff
@@ -45,4 +51,5 @@ dependencies {
     compile 'de.hdodenhof:circleimageview:2.0.0'
     compile 'com.facebook.android:facebook-android-sdk:[4,5)'
     compile 'com.google.android.gms:play-services:9.2.1'
+    compile 'com.android.support:multidex:1.0.1'
```
Comment (DerekFei) [inline app/build.gradle]: shouldn't change anything in gradle
For Linux environment, you may need:

sudo apt-get install libncurses5:i386 lib32z1 lib32ncurses5 zlib1g:i386

(see http://stackoverflow.com/questions/17005654/error-while-loading-shared-libraries-libncurses-so-5)

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -21,7 +21,8 @@
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:supportsRtl="true"
-        android:theme="@style/AppTheme">
+        android:theme="@style/AppTheme"
+        android:name="android.support.multidex.MultiDexApplication">
```
Comment (DerekFei) [inline app/src/main/AndroidManifest.xml]: shouldn't change manifest as well

File: app/src/main/java/com/packants/app/packants/Fragments/HistDetailsDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import android.view.Window;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.packants.app.packants.R;
+
+/**
+ * Created by Ben on 2016-07-23.
+ */
+public class HistDetailsDialogFragment extends DialogFragment{
```
Comment (0xABC) [inline app/src/main/java/com/packants/app/packants/Fragments/HistDetailsDialogFragment.java]: - I don’t think we gain anything by abbreviating History to Hist (use refactor tool to rename)
- need a space before {

File: app/src/main/java/com/packants/app/packants/Fragments/HistDetailsDialogFragment.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    private View mainView;
+    static HistDetailsDialogFragment getInstance() {
+        HistDetailsDialogFragment f = new HistDetailsDialogFragment();
+
+        // Supply num input as an argument.
```
Comment (0xABC) [inline app/src/main/java/com/packants/app/packants/Fragments/HistDetailsDialogFragment.java]: is this comment relevant? (below too)

**Comment by 0xABC** [comment]
As a rule I suggest using `this` wherever it makes sense (unless you’d have to prefix with the whole class name which looks too verbose). There are a lot of places where `this` is missing. (Alternatively use `m` prefix as is common in java but I’m not a fan of that approach)

**Comment by 0xABC** [comment]
Please rebase before merging.

**Pull request merged**